src/app/services/courses.service.ts

```diff
@@ -1,49 +1,62 @@
+import { HttpClient } from "@angular/common/http";
 import { Injectable } from "@angular/core";
 import { mockedAuthorsList } from "@app/shared/mocks/mocks";
+import { Observable } from "rxjs";
+import { newCourse } from "@app/shared/models/course.model";
+import { BaseUrl } from "@app/shared/constants/baseUrl";
 
 @Injectable({
   providedIn: "root",
 })
 export class CoursesService {
-  getAll() {
+  constructor(private http: HttpClient) {}
+
+  coursesApiUrl = `${BaseUrl}/courses`;
+  authorsApiUrl = `${BaseUrl}/authors`;
+  getAll(): Observable<any> {
     // Add your code here
+    return this.http.get<any>(`${this.coursesApiUrl}/all`);
   }
 
-  createCourse(course: any) {
+  createCourse(course: newCourse): Observable<any> {
     // replace 'any' with the required interface
     // Add your code here
+    return this.http.post<any>(`${this.coursesApiUrl}/add`, course);
```

Comment:
Instead of using any as type params, it would be better to define the actual types: course, user, etc.

Reply:
Please update with other occurrences in the service.